numpy/_core/src/multiarray/convert_datatype.c

```diff
@@ -344,7 +344,7 @@ PyArray_GetCastFunc(PyArray_Descr *descr, int type_num)
             PyObject *cobj;
 
             key = PyLong_FromLong(type_num);
-            cobj = PyDict_GetItem(obj, key);
+            cobj = PyDict_GetItem(obj, key); // noqa: borrowed-ref - manual fix needed
```

Comment:
Since this is a legacy function, could this be labeled as `\\ noqa: borrowed-ref OK`?